firmware/shared/src/io/io_fileSystem_logfs.c

```diff
@@ -221,12 +223,15 @@ FileSystemError io_fileSystem_close(int fd)
     return FILE_OK;
 }
 
-FileSystemError io_fileSystem_sync(int fd)
+FileSystemError io_fileSystem_sync(const int fd)
 {
-    if (!isValidFd(fd))
-    {
-        return FILE_ERROR;
-    }
-
+    CHECK_FILE_DESCRIPTOR(fd)
     return logfsErrorToFsError(logfs_sync(&fs, &files[fd]));
 }
+
+bool io_fileSystem_ready()
+{
+    // TODO recheck ready conditions
+    ready &= hw_sd_present();
+    return ready;
```

Comment:
Hmm I don't we should latch this. It's plausible that if we hit some bumps or something the SD card will rattle and the GPIO might blip not present. The IO CAN logging file keeps track of errors and only disables logging if we error 10 times.

Reply:
I think I am preserving old functionality, because before if the hw_sd_present is false at any point it stops the logging thread etc.